api/tools/src/main/java/org/pmiops/workbench/tools/LoadDataDictionary.java

```diff
@@ -41,41 +43,48 @@ public CommandLineRunner run(
 
       for (Resource resource : getDataDictionaryExportFiles()) {
         InputStream is = resource.getInputStream();
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
+        mapper.setDateFormat(df);
 
-        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
-        Timestamp definedTime = new Timestamp(Long.parseLong(reader.readLine()) * 1000);
+        mapper.registerModule(new KotlinModule());
+        DataDictionary dd = mapper.readValue(is, DataDictionary.class);
 
-        String line;
-        while ((line = reader.readLine()) != null) {
-          String[] fields = line.split(":");
+        Timestamp newEntryDefinedTime = dd.getMeta_data()[0].getCreated_time();
```

Comment:
I think you should guard against an empty metadata array here and below.

Reply:
I am OK with failing the script here if this throws an exception. It indicates a problem with the import file and it should be looked into before deploying.